From b4b8702d4907ea8b8857bb8beb5d345431bf2d07 Mon Sep 17 00:00:00 2001
From: Tai Huynh <tai.huynh.cp@renesas.com>
Date: Mon, 15 Sep 2025 14:46:27 +0700
Subject: [PATCH] Add: OpenCVA: and: Codec

Signed-off-by: Tai Huynh <tai.huynh.cp@renesas.com>
---
 include/configs/rzv2h-dev.h | 24 ++++++++++++++++++------
 1 file changed, 18 insertions(+), 6 deletions(-)

diff --git a/include/configs/rzv2h-dev.h b/include/configs/rzv2h-dev.h
index 534499430c..4de88a34f8 100644
--- a/include/configs/rzv2h-dev.h
+++ b/include/configs/rzv2h-dev.h
@@ -54,8 +54,12 @@
 	"prodsdbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk2p2 \0" \
 	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
 	"bootimage=booti 0x48080000 - 0x48000000 \0" \
-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a09g057h4-dev.dtb;run prodemmcbootargs \0" \
-	"sd2load=ext4load mmc 2:2 0x48080000 boot/Image;ext4load mmc 2:2 0x48000000 boot/r9a09g057h4-dev.dtb;run prodsdbootargs \0" \
+	"ocaaddr=0xA8000000 \0"     \
+	"ocabin=OpenCV_Bin.bin \0"  \
+	"codaddr=0xAFD00000 \0"     \
+	"codbin=Codec_Bin.bin \0"   \
+	"emmcload=ext4load mmc 0:2 ${ocaaddr} boot/${ocabin}; ext4load mmc 0:2 ${codaddr} boot/${codbin};ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a09g057h4-dev.dtb;run prodemmcbootargs \0" \
+	"sd2load=ext4load mmc 2:2 ${ocaaddr} boot/${ocabin}; ext4load mmc 2:2 ${codaddr} boot/${codbin}; ext4load mmc 2:2 0x48080000 boot/Image;ext4load mmc 2:2 0x48000000 boot/r9a09g057h4-dev.dtb;run prodsdbootargs \0" \
 	"bootcmd_check=if mmc dev 2; then run sd2load; else run emmcload; fi \0"
 #elif defined(CONFIG_TARGET_RZV2H_EVK_ALPHA)
 #define CFG_EXTRA_ENV_SETTINGS	\
@@ -65,8 +69,12 @@
 	"prodemmcbootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
 	"set_pmic=i2c dev 8; i2c mw 0x6a 0x22 0x0f; i2c mw 0x6a 0x24 0x00; i2c md 0x6a 0x00 0x30; i2c mw 0x12 0x8D 0x02; i2c md 0x12 0x20 0x80 \0" \
 	"bootimage=run set_pmic; booti 0x48080000 - 0x48000000 \0" \
-	"emmcload=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a09g057h4-evk-alpha.dtb;run prodemmcbootargs \0" \
-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a09g057h4-evk-alpha.dtb;run prodsdbootargs \0" \
+	"ocaaddr=0xA8000000 \0"     \
+	"ocabin=OpenCV_Bin.bin \0"  \
+	"codaddr=0xAFD00000 \0"     \
+	"codbin=Codec_Bin.bin \0"   \
+	"emmcload=ext4load mmc 0:2 ${ocaaddr} boot/${ocabin}; ext4load mmc 0:2 ${codaddr} boot/${codbin};ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a09g057h4-evk-alpha.dtb;run prodemmcbootargs \0" \
+	"sd1load=ext4load mmc 1:2 ${ocaaddr} boot/${ocabin}; ext4load mmc 1:2 ${codaddr} boot/${codbin}; ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a09g057h4-evk-alpha.dtb;run prodsdbootargs \0" \
 	"bootcmd_check=if mmc dev 1; then run sd1load; else run emmcload; fi \0"
 #else
 #define CFG_EXTRA_ENV_SETTINGS       \
@@ -75,8 +83,12 @@
 	"prodsd0bootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk0p2 \0" \
 	"prodsd1bootargs=setenv bootargs rw rootwait earlycon root=/dev/mmcblk1p2 \0" \
 	"bootimage=booti 0x48080000 - 0x48000000 \0" \
-	"sd0load=ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a09g057h44-rzv2h-evk.dtb;run prodsd0bootargs \0" \
-	"sd1load=ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a09g057h44-rzv2h-evk.dtb;run prodsd1bootargs \0" \
+	"ocaaddr=0xA8000000 \0"     \
+	"ocabin=OpenCV_Bin.bin \0"  \
+	"codaddr=0xAFD00000 \0"     \
+	"codbin=Codec_Bin.bin \0"   \
+	"sd0load=ext4load mmc 0:2 ${ocaaddr} boot/${ocabin}; ext4load mmc 0:2 ${codaddr} boot/${codbin};ext4load mmc 0:2 0x48080000 boot/Image;ext4load mmc 0:2 0x48000000 boot/r9a09g057h44-rzv2h-evk.dtb;run prodsd0bootargs \0" \
+	"sd1load=ext4load mmc 1:2 ${ocaaddr} boot/${ocabin}; ext4load mmc 1:2 ${codaddr} boot/${codbin}; ext4load mmc 1:2 0x48080000 boot/Image;ext4load mmc 1:2 0x48000000 boot/r9a09g057h44-rzv2h-evk.dtb;run prodsd1bootargs \0" \
 	"bootcmd_check=if mmc dev 1; then run sd1load; else run sd0load; fi \0"
 #endif
 
-- 
2.34.1

