diff -uprN old/arch/arm64/boot/dts/renesas/r9a09g057.dtsi new/arch/arm64/boot/dts/renesas/r9a09g057.dtsi
--- old/arch/arm64/boot/dts/renesas/r9a09g057.dtsi	2024-02-02 11:16:27.000000000 +0900
+++ new/arch/arm64/boot/dts/renesas/r9a09g057.dtsi	2024-02-02 11:42:44.000000000 +0900
@@ -1821,6 +1821,27 @@
 			status = "disabled";
 		};
 
+		opencva0: opencva@C0000000 {
+			compatible = "renesas,rzv2h-opencva";
+			reg = <0x00 0xC0000000 0x00 0x07D00000>;
+			status = "okay";
+		};
+
+		drp1: drp1@18000000 {
+			compatible = "renesas,rzv2h-drp";
+			reg = <0 0x18000000 0 0x1000000>;
+			interrupts = <GIC_SPI 912 IRQ_TYPE_LEVEL_HIGH>,
+				     <GIC_SPI 913 IRQ_TYPE_LEVEL_HIGH>;
+                        resets = <&cpg R9A09G057_DRP_ARESETN>;
+			status = "disabled";
+		};
+
+		codec0: codec {
+			reusable;
+			reg = <0x00 0xC7D00000 0x00 0x00300000>;
+			status = "okay";
+		};
+
 		rcar_sound: sound@13c00000 {
 			/*
 			 * #sound-dai-cells is required
diff -uprN old/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts new/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts
--- old/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts	2024-02-02 11:16:27.000000000 +0900
+++ new/arch/arm64/boot/dts/renesas/r9a09g057h4-dev.dts	2024-02-02 11:43:19.000000000 +0900
@@ -474,6 +474,13 @@
 	status = "okay";
 };
 
+&drp1 {
+    memory-region = <&codec0>;
+    memory-oca-region = <&opencva0>;
+
+    status = "okay";
+};
+
 &xhci0 {
 	pinctrl-0 = <&usb30_pins>;
 	pinctrl-names = "default";
diff -uprN old/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts new/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts
--- old/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts	2024-02-02 11:16:27.000000000 +0900
+++ new/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-alpha.dts	2024-02-02 11:43:58.000000000 +0900
@@ -1043,6 +1043,13 @@
 	status = "okay";
 };
 
+&drp1 {
+    memory-region = <&codec0>;
+    memory-oca-region = <&opencva0>;
+
+    status = "okay";
+};
+
 &canfd {
 	pinctrl-0 = <&can0_pins &can1_pins &can2_pins &can3_pins &can4_pins &can5_pins>;
 	pinctrl-names = "default";
diff -uprN old/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts new/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts
--- old/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts	2024-02-02 11:16:27.000000000 +0900
+++ new/arch/arm64/boot/dts/renesas/r9a09g057h4-evk-ver1.dts	2024-02-02 11:48:34.000000000 +0900
@@ -970,6 +970,13 @@
 	status = "okay";
 };
 
+&drp1 {
+    memory-region = <&codec0>;
+    memory-oca-region = <&opencva0>;
+
+    status = "okay";
+};
+
 &spi2 {
 	pinctrl-0 = <&spi2_pins>;
 	pinctrl-names = "default";
